### PR TITLE
eICU example

### PR DESCRIPTION
**Pull request by kingrc15**
- Created `run.sh` for running entire extraction pipeline
   - Reduced number of patients per shard to reduce shard sizes
 - Created eICU config file
 - Renamed columns for eICU v2.0

## Summary by CodeRabbit

- **New Features**
	- Introduced a new configuration file for extracting the eICU dataset, enhancing data processing capabilities.
	- Added a new script to streamline the processing of eICU data with defined steps and usage instructions.

- **Improvements**
	- Updated naming conventions for time-related columns and identifiers in configuration files for clarity and consistency.
	- Enhanced setup instructions in the README files for better user guidance on the MEDS extraction process.

### CODE REVIEW

**Comment by mmcdermott** [comment]
@kingrc15 is this ready for review, or should we convert it to a draft for now? I'm going through outstanding PRs to make sure I'm not blocking anything that needs my review. If it is ready for review, go ahead and add me as a reviewer and I'll take a look. Thanks!

File: eICU_Example/configs/extract_eICU.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+stage_configs:
+  shard_events:
+    infer_schema_length: 999999999
+
+stages:
+  - shard_events
+  - split_and_shard_subjects
+  - convert_to_sharded_events
+  - merge_to_MEDS_cohort
+  - extract_code_metadata
```
Comment (mmcdermott) [inline eICU_Example/configs/extract_eICU.yaml:34]: I think you could probably delete this stage (just delete the `  - extract_code_metadata` line) and everything would still work as I don't think you have any metadata extraction instructions in your config.

File: eICU_Example/run.sh [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+echo "the final unique check (which should not be necessary given the structure of eICU and is expensive) "
+echo "in the merge stage. You can do this by setting the following parameters at the end of the mandatory "
+echo "args when running this script:"
+echo "  * stage_configs.split_and_shard_subjects.n_subjects_per_shard=10000"
+echo "  * stage_configs.merge_to_MEDS_cohort.unique_by=null"
+echo "Additionally, consider reducing N_PARALLEL_WORKERS if > 1"
+
+echo "Running pre-MEDS conversion."
+./eICU_Example/pre_MEDS.py raw_cohort_dir="$EICU_RAW_DIR" output_dir="$EICU_PREMEDS_DIR"
+
```
Comment (mmcdermott) [inline eICU_Example/run.sh:53]: Everything after here should be deleted and instead we should use the one-line whole pipeline runner syntax like `MEDS_transform-runner "pipeline_config_fp=$PIPELINE_CONFIG_FP" "$@"` as in https://github.com/mmcdermott/MEDS_transforms/blob/dev/MIMIC-IV_Example/run.sh#L105

File: eICU_Example/configs/extract_eICU.yaml (hunk 1 above)
Comment (kingrc15) [reply eICU_Example/configs/extract_eICU.yaml]: This line was removed

**Review by mmcdermott — APPROVED** [review]

**Pull request merged**